### PR TITLE
[TEST BUILD] Installer image

### PR DESCRIPTION
**Pull request by pohly**
Includes also support for whole-disk encryption and dm-verity (currently limited to the installer image).

Whole-disk encryption depends on TPM 1.2 and recipes from meta-security with patches that are currently pending in meta-security master-next (for example, "trousers: tcsd.conf must be owned tss:tss"). The PR can be merged, but the feature can't be enabled.

There are some TODOs that depend on extending the product vs. development image concept. There was a discussion around that also in OE-core (triggered by the proposal to set a root password by default); we should reach some conclusion for OE-core and used that for our refkit-production.inc and 
refkit-development.inc files and also these TODO items. For now I propose to merge as-is.

Documentation only exists in the form of source code comments.

### CODE REVIEW

**Comment by pohly** [comment]
Question: should refkit-installer-image be created by default?

**Comment by pohly** [comment]
I suppose we want the installer image, so I've enabled it also by default.

**Comment by pohly** [comment]
Building multipath-tools fails - a known issue, the patch fixing it just isn't in refkit yet.

**Comment by pohly** [comment]
test this please

I've added the updated meta-openembedded with the multipath-tools fix.

File: meta-refkit/conf/distro/refkit.conf [hunk 1]
```diff
@@ -307,3 +307,7 @@ INHERIT += "uninative"
 # stage due to lack of RTC backup battery. This causes unnecessary fixes being
 # made due to filesystem metadata time stamps being in future.
 APPEND_append = " fsck.mode=skip"
+
+KERNEL_FEATURES_append = "${@ bb.utils.contains('DISTRO_FEATURES', 'dm-verity', ' dm-verity.scc', '', d) }"
+SRC_URI_append_pn-linux-yocto = " file://dm-verity.scc"
+FILESEXTRAPATHS_prepend_pn-linux-yocto = "${META_REFKIT_BASE}/files/linux-yocto:"
```
Comment (mythi) [inline meta-refkit/conf/distro/refkit.conf]: @pohly linux-intel is what we get from meta-intel nowadays. Have you submitted these in yocto-kernel-cache already? We've targeted to not have BSP bbappends here but all comes from meta-intel.

Comment (pohly) [inline meta-refkit/conf/distro/refkit.conf]: Yes, I have submitted it, but it's not yet available. The entire commit can be reverted once it is.

**Comment by pohly** [comment]
Please do not merge quite yet. While re-testing the dm-verity change without the local.conf changes that I was using earlier I am running into some problem with the verity.scc that I need to investigate first.

File: meta-refkit/conf/distro/refkit.conf (hunk 1 above)
Comment (pohly) [inline meta-refkit/conf/distro/refkit.conf]: Similar modifications for the new linux-intel need to be added.

Comment (mythi) [inline meta-refkit/conf/distro/refkit.conf]: It uses the same kernel cache. Once the scc is merged there let's submit a meta-intel patch ourselves to speed up things.

File: meta-refkit/classes/refkit-image.bbclass [hunk 2]
```diff
@@ -274,7 +274,11 @@ IMAGE_FSTYPES_remove = "live"
 # Activate "dsk" image type.
 IMAGE_CLASSES += "${@ 'image-dsk' if ${REFKIT_USE_DSK_IMAGES} else ''}"
 
+# By default, the full image is going to use roughly 4GB, independent
+# of the actual roofs size.
 WKS_FILE = "refkit-directdisk.wks.in"
+REFKIT_VFAT_MB ??= "30"
```
Comment (jlaako) [inline meta-refkit/classes/refkit-image.bbclass]: I recommend using 32

File: meta-refkit/classes/refkit-image.bbclass [hunk 3]
```diff
@@ -274,7 +274,11 @@ IMAGE_FSTYPES_remove = "live"
...
+REFKIT_IMAGE_SIZE ??= "--fixed-size 3700M"
```
Comment (jlaako) [inline meta-refkit/classes/refkit-image.bbclass]: And 3968 here

**Pull request closed without merging**